### PR TITLE
Improve WAL/mmap chunks test for histograms

### PR DESCRIPTION
**Pull request by codesome**
NOTE: this is for sparsehistogram branch.

This PR build on top of https://github.com/prometheus/prometheus/pull/11194

This PR improves unit test to take care of all these points from https://github.com/prometheus/prometheus/issues/11171.

	- [ ] WAL/M-map
		- [ ] Writing to WAL
		- [ ] Reading from WAL
		- [ ] M-mapping histogram chunks
		- [ ] Replay of data on restart
			- [ ] Series having only histograms
				- [ ] With m-map chunks
			- [ ] Series having a mix of histograms and float
				- [ ] With m-map chunks having both histograms and float

TSDB does not support histograms in checkpointing of WAL (which is one of the test case specified in #11171). I will take care of it in a follow up PR.

### CODE REVIEW

File: tsdb/record/record.go [hunk 1]
```diff
@@ -44,10 +44,10 @@ const (
 	Tombstones Type = 3
 	// Exemplars is used to match WAL records of type Exemplars.
 	Exemplars Type = 4
-	// Histograms is used to match WAL records of type Histograms.
-	Histograms Type = 5
 	// Metadata is used to match WAL records of type Metadata.
 	Metadata Type = 6
+	// Histograms is used to match WAL records of type Histograms.
+	Histograms Type = 7
```
Comment (beorn7) [inline tsdb/record/record.go:50]: Why this change?

**Review by beorn7 — COMMENTED** [review]
Looks good, but I don't understand the `Type` enum change in tsdb/record/record.go

File: tsdb/record/record.go (hunk 1 above)
Comment (codesome) [reply tsdb/record/record.go:50]: Oh I forgot to mention. So the backstory is that the out of order support brings in a new record type that is already using the number 5. Given it is in production at places already, I think it is better to change this one here since native histograms are not anywhere (except one instance in the world).
